### PR TITLE
Router LoginRestricted and analytics core components cleanup

### PR DESCRIPTION
**Pull request by goran-peoski-work**
In general there is more work to be done on the hooks and Reactfire, 
but to avoid major issues, 
the decision is to add chunks of the merged-then-reverted PR 
- https://github.com/sparkletown/sparkle/pull/2894

piecemeal in order to catch culprits.

### CODE REVIEW

File: src/components/templates/Auditorium/components/SeatingBlock/SeatingBlock.tsx [hunk 1]
```diff
@@ -44,7 +44,7 @@ export const SeatingBlock: React.FC<SeatingBlockProps> = ({ space }) => {
 
   return (
     <Switch>
-      <Section venue={space} sectionId={sectionId} user={user} />
```
Comment (colinhowe) [inline src/components/templates/Auditorium/components/SeatingBlock/SeatingBlock.tsx:47]: We only allow for one section at the moment, this change isn't necessary

**Review by colinhowe — CHANGES_REQUESTED** [review]
Minor change and then good to go

File: src/components/templates/Auditorium/components/SeatingBlock/SeatingBlock.tsx (hunk 1 above)
Comment (goran-peoski-work) [reply src/components/templates/Auditorium/components/SeatingBlock/SeatingBlock.tsx:47]: That's a leftover from before, no? It kind of snuck in from earlier

Comment (goran-peoski-work) [reply src/components/templates/Auditorium/components/SeatingBlock/SeatingBlock.tsx:47]: What change?

File: src/components/templates/Auditorium/components/SeatingBlock/SeatingBlock.tsx [hunk 2]
```diff
@@ -44,7 +44,7 @@ export const SeatingBlock: React.FC<SeatingBlockProps> = ({ space }) => {
...
+      {sectionId && <Section venue={space} sectionId={sectionId} user={user} />}
```
Comment (colinhowe) [inline src/components/templates/Auditorium/components/SeatingBlock/SeatingBlock.tsx:47]:
```suggestion
      {<Section venue={space} sectionId={sectionId} user={user} />}
```

File: src/components/templates/Auditorium/components/SeatingBlock/SeatingBlock.tsx (hunk 1 above)
Comment (goran-peoski-work) [reply src/components/templates/Auditorium/components/SeatingBlock/SeatingBlock.tsx:47]: Regardless what other places of the code and the DB have, according to the line above:

`const sectionId = urlSectionId || sortedSections[0]?.id;` 

And according to the fact I did have page breaking error without the check above and the one on this line, I think it's better safe than sorry.

**Review by colinhowe — APPROVED** [review]

**Pull request merged**